### PR TITLE
New SVG component for truncated text with tooltip, and application to survival chart legend

### PR DESCRIPTION
**Pull request by adamabeshouse**
https://www.cbioportal.org/comparison/survival?sessionId=5cf13d70e4b0ab4137874293
image

downloaded SVG:

image

### CODE REVIEW

File: src/shared/components/TruncatedTextSVG.tsx [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+            <>
+                <text
+                    {...rest}
+                    onMouseOver={this.onMouseOver}
+                    onMouseOut={this.onMouseOut}
+                    onMouseMove={this.onMouseMove}
+                >
+                    {this.textReport.text}
+                </text>
+                {this.textReport.truncated && this.tooltipOpen && (ReactDOM as any).createPortal(
```
Comment (alisman) [inline src/shared/components/TruncatedTextSVG.tsx]: we can use react-portal library to avoid imperative call to creatPortal.  e.g. <Portal>

Comment (adamabeshouse) [reply src/shared/components/TruncatedTextSVG.tsx]: okay cool

File: src/pages/resultsView/survival/SurvivalChart.tsx [hunk 2]
```diff
@@ -300,7 +307,8 @@ export default class SurvivalChart extends React.Component<ISurvivalChartProps,
 
     @autobind
     private getSvg() {
-        return this.svgContainer.firstChild;
+        return this.toSVGDOMNode();
+        //return this.svgContainer.firstChild;
```
Comment (zhx828) [inline src/pages/resultsView/survival/SurvivalChart.tsx]: We can get rid of this now?

File: src/shared/components/TruncatedTextSVG.tsx [hunk 3]
```diff
@@ -0,0 +1,118 @@
+import * as React from "react";
```
Comment (alisman) [inline src/shared/components/TruncatedTextSVG.tsx]: should we add tooltip to the name of this component?  also, would it make sense to somehow combine this with victory label so we could use it elsewhere?

**Comment by alisman** [comment]
The ellipsis are right up against the right edge of chart making them still a little hard to make out.   Lets tweak to get it right.

File: src/shared/components/TruncatedTextSVG.tsx (hunk 3 above)
Comment (adamabeshouse) [reply src/shared/components/TruncatedTextSVG.tsx]: yeah, this is victory compatible. it's a plug in as `labelComponent`

**Comment by adamabeshouse** [comment]
@alisman @zhx828 made requested changes

**Pull request merged**